### PR TITLE
feat: create `no-restricted-jest-methods` rule

### LINKED ISSUE #1256: New rule to disallow using mock API in end-to-end test files
What about adding a rule which would disallows using mock APIs? I though it could be activated for a directory where end-to-end tests live and could ensure that test fixtures are used instead of mocks.

Usually I prefer to leave mocks and spies for unit tests and to avoid them in end-to-end tests. Just wanted to check what people smarter than me think? (;

### PR DESCRIPTION
**Pull request by G-Rath**
Resolves #1256

### CODE REVIEW

**Comment by mrazauskas** [comment]
That was quick. Thanks!

**Review by SimenB — APPROVED** [review]
great stuff - very clean implementation 👍

File: docs/rules/no-restricted-jest-methods.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+`all`.
+
+<!-- end rule header -->
+
+You may wish to restrict the use of specific `jest` methods.
+
+## Rule details
+
+This rule checks for the usage of specific methods on the `jest` object, which
+can be used to disallow curtain patterns such as spies and mocks.
```
Comment (Primajin) [inline docs/rules/no-restricted-jest-methods.md:14]: just for my non native speaker understanding - are we really talking about `curtain patterns` here or is this a typo of `certain patterns`?

Comment (SimenB) [reply docs/rules/no-restricted-jest-methods.md:14]: typo 😀 

wanna send a PR fixing it?

Comment (Primajin) [reply docs/rules/no-restricted-jest-methods.md:14]: More than happy to, coming right up

Comment (Primajin) [reply docs/rules/no-restricted-jest-methods.md:14]: https://github.com/jest-community/eslint-plugin-jest/pull/1259

**Pull request merged**